### PR TITLE
Fix: notification panel fixes

### PR DESCRIPTION
**Pull request by hellypi**

### CODE REVIEW

**Comment by hellypi** [comment]
[Notification panel fixes](https://trello.com/c/jNAKCvQD/576-notification-panel-fixes)

File: new-lamassu-admin/src/components/NotificationCenter/NotificationRow.js [hunk 1]
```diff
@@ -73,8 +73,10 @@ const NotificationRow = ({
         </Grid>
       </Grid>
       <Grid item xs={3} style={{ zIndex: 1 }}>
-        {!read && (
-          <div onClick={() => onClear(id)} className={classes.unreadIcon} />
+        {read ? (
+          <div onClick={() => toggleClear(id)} className={classes.readIcon} />
+        ) : (
+          <div onClick={() => toggleClear(id)} className={classes.unreadIcon} />
```
Comment (hellypi) [inline new-lamassu-admin/src/components/NotificationCenter/NotificationRow.js]: change this to how it's done in percentageGraph

File: new-lamassu-admin/src/components/layout/Header.js [hunk 2]
```diff
@@ -56,15 +57,27 @@ const Subheader = ({ item, classes }) => {
...
+  const notifCenterButtonRef = useRef()
+  const popperRef = useRef()
   const history = useHistory()
   const classes = useStyles()
 
+  useEffect(() => {
+    if (data?.hasUnreadNotifications) return setHasUnread(true)
+    // if not true, make sure it's false and not undefined
+    if (notNil(data?.hasUnreadNotifications)) return setHasUnread(false)
+  }, [data])
```
Comment (chaotixkilla) [inline new-lamassu-admin/src/components/layout/Header.js:79]: I presume this is for the bell icon on the header being lit up or not. Not the biggest fan of this approach, I think you can reach the same result without using the useEffect hook, but I'll pass it and if Rafael finds it alright, so it is

**Review by chaotixkilla — APPROVED** [review]

File: new-lamassu-admin/src/components/NotificationCenter/NotificationCenter.styles.js [hunk 3]
```diff
@@ -34,19 +34,20 @@ const styles = {
   },
   actionButtons: {
     display: 'flex',
-    marginLeft: spacer * 2
+    marginLeft: spacer * 2,
+    height: 0
   },
-  notificationIcon: {
+  notificationIcon: ({ notifButtonCoords, xOffset }) => ({
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/components/NotificationCenter/NotificationCenter.styles.js]: migh as well just call this variable buttonCoords

File: new-lamassu-admin/src/components/NotificationCenter/NotificationCenter.styles.js [hunk 4]
```diff
@@ -80,13 +81,24 @@ const styles = {
...
     zIndex: 1
   },
+  readIcon: {
+    marginLeft: spacer,
+    marginTop: 5,
+    width: '12px',
+    height: '12px',
+    border: `1px solid ${comet}`,
+    borderRadius: '50%',
+    cursor: 'pointer',
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/components/NotificationCenter/NotificationCenter.styles.js]: You can replace this by non text versions:

```js
{
  width: 12,
  height: 12,
  border: [[1, 'solid', comet]]
 } 
```

File: new-lamassu-admin/src/components/NotificationCenter/NotificationRow.js (hunk 1 above)
Comment (RafaelTaranto) [reply new-lamassu-admin/src/components/NotificationCenter/NotificationRow.js]: Is this still relevant? Also might as well just change the className based on the variable:

```js
const clearClasses = {
  [classes.readIcon]: read,
  [classes.unreadIcon]: !read
}
```

File: new-lamassu-admin/src/components/layout/Header.js (hunk 2 above)
Comment (RafaelTaranto) [reply new-lamassu-admin/src/components/layout/Header.js:79]: Yeah, since hasUnread becase a state variable I don't think there's a way out of this

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]

File: new-lamassu-admin/src/components/NotificationCenter/NotificationCenter.js [hunk 5]
```diff
@@ -51,21 +52,31 @@ const CLEAR_ALL_NOTIFICATIONS = gql`
...
-
   const notifications = R.path(['notifications'])(data) ?? []
-  const hasUnread = data?.hasUnreadNotifications ?? false
-  if (!hasUnread) {
-    notifyUnread && notifyUnread()
-  }
-  const [clearNotification] = useMutation(CLEAR_NOTIFICATION, {
+  const [hasUnread, setHasUnread] = useState(
+    hasUnreadProp || (data?.hasUnreadNotifications ?? false)
+  )
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/components/NotificationCenter/NotificationCenter.js]: I don't think there is a scenario where data is filled and this runs, is there?

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]

**Review by RafaelTaranto — APPROVED** [review]

**Pull request merged**